src/EventGridEmulator/EventHandling/BaseEventHttpContextHandler.cs

```diff
@@ -31,22 +34,23 @@ public async Task HandleAsync(HttpContext context, string topic)
 
     private async Task<IResult> HandleInternalAsync(HttpContext context, string topic)
     {
-        if (!this._options.CurrentValue.Topics.TryGetValue(topic, out var subscribers))
-        {
-            return Results.Ok();
-        }
-
         var events = await EventsSerializer.DeserializeEventsAsync<TEvent>(context);
         if (events == null)
         {
             return Results.BadRequest();
         }
 
-        foreach (var subscriber in subscribers)
+        var pushSubscriber = this._options.CurrentValue.GetPushSubscribers(topic);
+        foreach (var subscriber in pushSubscriber)
         {
-            var cancellationToken = this._cancellationTokenRegistry.Get(topic, subscriber);
+            var cancellationToken = this._cancellationTokenRegistry.Get(topic, subscriber.Uri);
             this.EnhanceEventData(events, topic);
```

Comment:
Good point, I think this doesn't need to be done in each loop iteration either, since the events are obtained from httpcontext and topic is passed to the method; previously we were running the same eventdata enhancement for each subscriber

Reply:
EnhancingEventData is setting a default topic value to our as topic